raiden/blockchain/events.py

```diff
@@ -285,22 +285,25 @@ class BlockchainEvents:
 
     def __init__(self):
         self.event_listeners = list()
-        self.first_run = True
 
     def reset(self):
-        self.first_run = True
+        listeners = [
+            event_listener._replace(first_run=True)
+            for event_listener in self.event_listeners
+        ]
+        self.event_listeners = listeners
 
     def poll_blockchain_events(self):
         # When we test with geth if the contracts have already been deployed
         # before the filter creation we need to use `get_all_entries` to make
         # sure we get all the events. With tester this is not required.
-        if self.first_run:
-            query_fn = 'get_all_entries'
-            self.first_run = False
-        else:
-            query_fn = 'get_new_entries'
 
         for event_listener in self.event_listeners:
+            query_fn = 'get_new_entries'
+            if event_listener.first_run is True:
```

Comment:
argh .. damn this is getting uglier and uglier :(

I am the one who first introduced this `self.first_run` thing due to what the comment describes.

We should really find out if this is a bug or normal behaviour from geth. I would expect that a new filter asking `get_new_entries()` would get all entries, even if they predate its creation.

Reply:
> We should really find out if this is a bug or normal behaviour from geth. I would expect that a new filter asking get_new_entries() would get all entries, even if they predate its creation.

I've traced RPC calls and in some situations filter just doesn't return the event. It might be a some obscure bug in geth - since it's very hard to reproduce I'd opt in for using `get_all_entries` as it reliably returns expected events.
We can investigate this buggy behaviour and file a bug report to `geth`, but I guess it's not a priority now if we have something that works.

Reply:
@LefterisJP and @pcppcp which builds/logs do we have to show this behavior?

Reply:
https://travis-ci.org/pcppcp/raiden/jobs/399518063

search for `ContractReceiveChannelNewBalance.*d34dd` and `eth_getfilterchanges.*d34dd.*topics.*0x61cf`

Reply:
For me this happens also reliably locally when the tests created the geth filter after the block the event was fired and indeed the `get_new_entries` was not returning the events before the filter creation.

But anyhow I agree with jozef. When we have less work to do, we can talk with the geth people, perhaps file a bug report and improve the situation.